Sources/Signal.swift

```diff
@@ -2422,6 +2422,21 @@ extension Signal where Error == NoError {
 		}
 	}
 
+	/// Promote a signal that does not generate failures into one that can.
+	///
+	/// - note: This does not actually cause failures to be generated for the
+	///         given signal, but makes it easier to combine with other signals
+	///         that may fail; for example, with operators like
+	///         `combineLatestWith`, `zipWith`, `flatten`, etc.
+	///
+	/// - parameters:
+	///   - _ An `ErrorType`.
+	///
+	/// - returns: A signal that has an instantiatable `ErrorType`.
+	public func promoteError(_: Error.Type = Error.self) -> Signal<Value, Error> {
```

Comment:
Perhaps I should be more careful about my wordings. Nothing is silenced here, and the overload does not enable anything new. It already works today, and the overload is just a fast path when it knows `Self.Error` is the `Error` of the return producer.